### PR TITLE
Combo Jito txs (using bundles) + Performance callback

### PR DESCRIPTION
**Pull request by filipzeta**
- Add Jito bundles functionality
- Allow the choice between just normal RPC sending (includes Jito /transactions endpoint), just Jito bundle or both
- Add performance callback to Exchange
- Add the option to simulate transactions to more efficiently choose the compute unit limit for better tx success

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -951,220 +952,54 @@ function txConfirmationCheck(expectedLevel: string, currentLevel: string) {
...
-    method: "sendTransaction",
-    params: [encodedTx],
+    method: "getBundleStatuses",
+    params: [[bundle]],
   };
 
-  let txOpts = opts || commitmentConfig(provider.connection.commitment);
-  let txSig: string;
   try {
+    console.log("[DEBUG] sending getBundleStatuses Jito request");
```
Comment (kelvinlau20100) [inline src/utils.ts:965]: old code here? or maybe we could just do console.debug()

File: src/utils.ts [hunk 2]
```diff
@@ -951,220 +952,54 @@ function txConfirmationCheck(expectedLevel: string, currentLevel: string) {
...
     const response = await axios.post(jitoURL, payload, {
       headers: { "Content-Type": "application/json" },
     });
-    txSig = response.data.result;
+    console.log("[DEBUG] sent getBundleStatuses Jito request");
+    if (response.status != 200) {
+      throw Error(`Bundle ${bundle} not accepted by Jito`);
```
Comment (kelvinlau20100) [inline src/utils.ts:971]: maybe this error should be bundle status not found on jito

File: src/utils.ts [hunk 3]
```diff
@@ -951,220 +952,54 @@ function txConfirmationCheck(expectedLevel: string, currentLevel: string) {
...
+    }
+    return response.data.result;
+    // console.log(response.data.result);
+    // let res = response.data.result;
```
Comment (kelvinlau20100) [inline src/utils.ts:975]: old commentted code?

File: src/utils.ts [hunk 4]
```diff
@@ -1180,21 +1015,58 @@ export async function sendRawTransactionCaught(con: Connection, rawTx: any) {
...
+      paramsTxs.push(bs58.encode(tx));
+    }
+    const payload = {
+      jsonrpc: "2.0",
+      id: 1,
+      method: "sendBundle",
+      params: [paramsTxs],
+    };
+
+    console.log("[DEBUG] sending to Jito /bundles endpoint");
```
Comment (kelvinlau20100) [inline src/utils.ts:1055]: debug logs

**Comment by filipzeta** [comment]
Closing this for now to minimise accidentally breaking the SDK because it touches a lot of core tx sending code and it's only nice to have. If FE work resumes on this then I can re-open it.

**Pull request closed without merging**